website/docs/dev/cluster/migration.md

```diff
@@ -0,0 +1,310 @@
+---
+id: slot-migration
+sidebar_label: Migration
+title: Migration Overview
+---
+
+# Slot Migration Overview
+
+In Garnet, slot migration, refers to the act of re-assigning slot ownernship and moving (i.e. ***migrating***) the associated key-values pairs from one node to another while the cluster is fully operational.
+The migration operation is only available in cluster mode.
+It can be initiated only by the original owner (a.k.a source node) of a given slot and addressed towards another already known, and trusted cluster node (a.k.a target node).
+Both the source and target node have to be primary nodes.
+Migration is triggered manually by using the MIGRATE command that operates in two modes (1) Migrate individual keys (2) Migrate entire slots or range of slots.
+This page is focused on the slot migration implementation details.
+For more information about using the associated command refer to the slot migration [user guide](../../cluster/key-migration).
+
+
+# Implementation Details
+
+The implementation of migration operation is separated into two components
+
+1. Command parsing and validation component.
+2. Migrate session operation and management component.
+
+
+The first component is responsible for parsing and validating the migration command arguments.
+Validation involves the following:
+
+1. Parse every migration option and validate arguments associated with that option.
+2. Validate ownership of associated slot that is about to be migrate.
+3. For KEYS option, validate that keys to be migrated do not hash across different slots.
+4. For SLOTS/SLOTSRANGE option, validate no slot is referenced multiple times.
+5. For SLOTSRANGE option, make sure a range is defined by a pair of values.
+6. Validate that target of migration is known, trusted and has the role of a primary.
+
+When parsing completes succesfully, a migrate session is created and executed.
+Depending on the chosen option, the migrate session executes as a foreground task (using KEYS option) or a background task (SLOTS/SLOTSRANGE option).
+
+The second component is separated into the following sub-components:
+
+1. ```MigrationManager```
+2. ```MigrateSessionTaskStore```
+3. ```MigrateSession```
+
+The ```MigrationManager``` is responsible for managing the active ```MigrateSession``` tasks.
+It uses ```MigrateSessionTaskStore``` to atommically add or remove new ```MigrateSession```.
```

Comment:
Typo:

```suggestion
It uses ```MigrateSessionTaskStore``` to atomically add or remove new ```MigrateSession```.
```

Reply:
Good catch, will have to enable spell check in vs code.